bin/exec.ml

```diff
@@ -93,18 +93,36 @@ let term =
         let prog = Path.extend_basename prog ~suffix:Bin.exe in
         Option.some_if (Path.exists prog) prog
   in
+  (* Good candidates for the "./x.exe" instead of "x.exe" error are executables
+     present at the root of the build dir *)
+  let hints =
+    lazy
```

Comment:
`let hints () = ...` will work just as well here since `hints` is used only once in each case.